### PR TITLE
fix(@angular-devkit/build-angular): handle base tag endOffset in inde…

### LINKED ISSUE #18381: Build with baseHref removes opening tag of the next HTML element
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

Not sure.

### Description

ng build produces invalid HTML

## 🔬 Minimal Reproduction

1. Add `<base href="/" /></head>` to index.html. **There should be no space between closing and opening tags** (I run HTML minification before the build step)
2. Use i18n (I am not sure if this is necessary)
3. Run `ng build --localize`

## 🔥 Exception or Error
<pre>

Updated index.html has: &lt;base href="/app/en/">/head>
instead of &lt;base href="/app/en/">&lt;/head>
</pre>

## 🌍 Your Environment
<pre><code>

Angular CLI: 10.0.4
Node: 14.4.0
OS: win32 x64

Angular: 10.0.5
... animations, common, compiler, compiler-cli, core
... language-service, localize, platform-browser
... platform-browser-dynamic, router, service-worker
Ivy Workspace: <error>

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.1000.4
@angular-devkit/build-angular     0.1000.4
@angular-devkit/build-optimizer   0.1000.4
@angular-devkit/build-webpack     0.1000.4
@angular-devkit/core              10.0.4
@angular-devkit/schematics        10.0.4
@angular/cli                      10.0.4
@angular/pwa                      0.1000.4
@ngtools/webpack                  10.0.4
@schematics/angular               10.0.4
@schematics/update                0.1000.4
rxjs                              6.6.0
typescript                        3.9.7
webpack                           4.44.0
</code></pre>

**Anything else relevant?**

### PR DESCRIPTION
**Pull request by Ayudh**
…x.html generation

* fix endOffset if the base tag is already present
* add test case for this

Closes #18381

### CODE REVIEW

File: packages/angular_devkit/build_angular/src/angular-cli-files/utilities/index-file/augment-index-html_spec.ts [hunk 1]
```diff
@@ -49,6 +49,24 @@ describe('augment-index-html', () => {
     `);
   });
 
+  it('should emit correct when base-href is present', async () => {
```
Comment (alan-agius4) [inline packages/angular_devkit/build_angular/src/angular-cli-files/utilities/index-file/augment-index-html_spec.ts:52]: Maybe `should replace base href when index content is one liner`?

Comment (Ayudh) [reply packages/angular_devkit/build_angular/src/angular-cli-files/utilities/index-file/augment-index-html_spec.ts:52]: This test case is when base tag is present in `index.html`. How about `should replace base href value`?

**Review by alan-agius4 — CHANGES_REQUESTED** [review]
Thanks for this. 

Seems like the tests failure is because https://github.com/angular/angular-cli/blob/e11d55679e5bfcf55aef25d060963deecdb4fb90/packages/angular_devkit/build_angular/src/browser/specs/service-worker_spec.ts#L238 needs to be changed to the new hash `e92a780dba6f6099adafb72d7611f6660a91a70b `

File: packages/angular_devkit/build_angular/src/angular-cli-files/utilities/index-file/augment-index-html_spec.ts (hunk 1 above)
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/angular-cli-files/utilities/index-file/augment-index-html_spec.ts:52]: SGTM

**Comment by Ayudh** [comment]
> Thanks for this.
> 
> Seems like the tests failure is because
> 
> https://github.com/angular/angular-cli/blob/e11d55679e5bfcf55aef25d060963deecdb4fb90/packages/angular_devkit/build_angular/src/browser/specs/service-worker_spec.ts#L238
> 
> needs to be changed to the new hash `e92a780dba6f6099adafb72d7611f6660a91a70b `

Changing this file will fix the test cases? Any docs to understand the specs here?

**Comment by alan-agius4** [comment]
See: https://angular.io/guide/service-worker-intro#service-workers-in-angular

**Comment by Ayudh** [comment]
@alan-agius4 I am not sure how to fix the test cases. Can you suggest?

**Comment by alan-agius4** [comment]
@Ayudh, seem like a flake, I'll re-run the CI step.

**Review by alan-agius4 — APPROVED** [review]
Thanks, LGTM

**Pull request merged**